src/user/views/adsManager/views/advanced/components/campaign/fields/FormatField.tsx

```diff
@@ -39,23 +39,22 @@ const FormatItemButton = (props: { format: CampaignFormat } & PriceProps) => {
   const { isEdit } = useIsEdit();
   const [, meta, format] = useField<CampaignFormat>("format");
   const [, , price] = useField<string>("price");
-  const [, bMeta, billing] = useField<Billing>("billingType");
+  const [, billingType, helper] = useField<Billing>("billingType");
 
   return (
     <ListItemButton
       disabled={isEdit}
       selected={meta.value === props.format}
       onClick={() => {
         format.setValue(props.format);
-        const found = props.prices.find((p) => {
-          return (
+        const found = props.prices.find(
+          (p) =>
             p.format === props.format &&
-            p.billingType === bMeta.value.toUpperCase()
-          );
-        });
+            p.billingType === billingType.value.toUpperCase(),
+        );
         if (props.format === CampaignFormat.NewsDisplayAd) {
           price.setValue(found?.billingModelPrice ?? "10");
-          billing.setValue("cpm");
+          if (billingType.value === "cpc") helper.setValue("cpm");
```

Comment:
This looks wrong?

Oh. If the billing type was cpc before the format switched to news, then force the billing type to `cpm` is I think the behaviour. But that doesn't entirely make sense to me: surely if the price changes the billing model also has to change, since the two are linked - a CPC price doesn't make sense for CPM.